python/tests/test_projects.py

```diff
@@ -175,16 +176,55 @@ def test_delete_project_should_delete_project(
     assert projects_dict == {}
 
 
-def test_get_project_files_should_return_list_of_files(
-    monkeypatch, tmp_path, setup_project_path_storage
-):
+def test_get_project_files(monkeypatch, tmp_path, setup_project_path_storage):
     monkeypatch.setattr(projects.settings, "WEB_STORAGE_URL", tmp_path)
 
     user_id = "user1"
     project_id = "project1"
 
+    # setup project files
     project_path = projects.get_project_path(user_id, project_id)
-    (Path(project_path) / "file1.txt").touch()
+    project_name = projects.get_project_name(user_id, project_id)
+
+    # hidden dirs/files should be ignored
+    (Path(project_path) / "file1.refstudio").touch()
+    (Path(project_path) / ".ignoreme").touch()
+    project_path.joinpath("uploads").mkdir()
+    (Path(project_path) / "uploads" / "file2.pdf").touch()
+    project_path.joinpath(".storage").mkdir()
+    (Path(project_path) / ".storage" / "references.json").touch()
+    project_path.joinpath("empty").mkdir()
+
+    response = projects.get_project_files(user_id, project_id)
+
+    expected = {
+        "id": project_id,
+        "name": project_name,
+        "path": str(project_path),
+        "contents": [
+            {
+                "name": "file1.refstudio",
+                "path": "file1.refstudio",
+                "file_extension": ".refstudio",
+            },
+            {
+                "name": "empty",
+                "path": "empty",
+                "children": [],
+            },
+            {
+                "name": "uploads",
+                "path": "uploads",
+                "children": [
+                    {
+                        "name": "file2.pdf",
+                        "path": "uploads/file2.pdf",
+                        "file_extension": ".pdf",
+                    }
+                ],
+            },
+        ],
+    }
 
-    files = projects.get_project_files(user_id, project_id)
-    assert files == [Path(f"{project_path}/file1.txt")]
+    assert isinstance(response, ProjectFileTreeResponse)
+    assert sorted(response.dict()) == sorted(expected)
```

Comment:
This doesn't quite work, `sorted(dict)` sorts whatever you'd get if you iterated over `dict`, which in this case is just the keys. So `sorted(expected) == ['contents', 'id', 'name', 'path']`.

I think the right way to handle this is to have `get_project_files` always return children in alphabetical order.

Reply:
We can handle sort in the client code. Maybe that's what makes most sense. CC @sehyod

Reply:
Agree that client side handling sort makes the most sense.